lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -764,12 +780,23 @@ def _flip(self):
         flipped_kymo = self._kymo.flip()
         return KymoTrackGroup([track._flip(flipped_kymo) for track in self])
 
+    def _split_track(self, track, split_node):
+        """Split a track at a particular node
+
+        Parameters
+        ----------
+        track : KymoTrack
+            Track to split
+        split_node : int
+            Index of the node in the track to split at
+        """
+        first_half, second_half = track._split(split_node)
+        self._src[self._src.index(track)] = first_half
+        self._src.append(second_half)
```

Comment:
I wonder would it make more sense to insert the "new" track right after the first half of the original one? Basically if you start with `[track1, track2, track3]` and split on `track2` end up with `track1, track2a, track2b, track3]`. I think there are arguments for both. This way you still preserve the spatial order of the tracks (but that's not really guaranteed on our side anyway once you start messing the things), whereas your proposal would preserve the indexing of the list.

I could be convinced either way, just some food for thought

Reply:
Personally, I see more room for error by performing that insertion here. One might not realize they're performing an insertion, while iterating on a list and selectively splitting and then get errors.

Reply:
good point, yeah let's keep it as is